voidcrew/code/game/objects/items/devices/walkman/cassette.dm

```diff
@@ -0,0 +1,217 @@
+	var/side2_icon = "cassette_worstmap"
+	///if the cassette is flipped, for playing second list of songs
+	var/flipped = FALSE
+	///list of songs each side has to play
+	var/list/songs = list("side1" = list(),
+						  "side2" = list())
+	///list of each songs name in the order they appear
+	var/list/song_names = list("side1" = list(),
+						 	   "side2" = list())
+	///the id of the cassette
+	var/id = 1
+	//the cassette_tape type datum
+	var/datum/cassette/cassette_tape/tape
+
+/obj/item/device/cassette_tape/Initialize()
+	. = ..()
+	tape = new tape
+	name = tape.name
+	desc = tape.desc
+	icon_state = tape.icon_state
+	side1_icon = tape.side1_icon
+	side2_icon = tape.side2_icon
+	id = tape.id
+	songs = tape.songs
+	song_names = tape.song_names
+	qdel(tape)
+
+/obj/item/device/cassette_tape/attack_self(mob/user)
+	..()
+
+	icon_state = flipped ? side1_icon : side2_icon
+	flipped = !flipped
+	to_chat(user,"You flip [src]")
+
+/obj/item/device/cassette_tape/attackby(obj/item/item, mob/living/user)
+	if(!istype(item, /obj/item/pen))
+		return
+	var/choice = input("What would you like to change?") in list("Cassette Name", "Cassette Description", "Cancel")
+	switch(choice)
+		if("Cassette Name")
+			///the name we are giving the cassette
+			var/newcassettename = reject_bad_text(stripped_input(user, "Write a new Cassette name:", name, name))
+			if(!user.canUseTopic(src, BE_CLOSE))
+				return
+			if (length(newcassettename) > 20)
+				to_chat(user, "<span class='warning'>That name is too long!</span>")
+				return
+			if(!newcassettename)
+				to_chat(user, "<span class='warning'>That name is invalid.</span>")
+				return
+			else
+				name = "[lowertext(newcassettename)]"
+		if("Cassette Description")
+			///the description we are giving the cassette
+			var/newdesc = stripped_input(user, "Write a new description:", name, desc)
+			if(!user.canUseTopic(src, BE_CLOSE))
+				return
+			if (length(newdesc) > 180)
+				to_chat(user, "<span class='warning'>That description is too long!</span>")
+				return
+			if(!newdesc)
+				to_chat(user, "<span class='warning'>That description is invalid.</span>")
+				return
+			else
+				desc = newdesc
+		else
+			return
+
+/datum/cassette/cassette_tape
+	var/name = "Broken Cassette"
+	var/desc = "You shouldn't be seeing this! Make an issue about it"
+	var/icon_state = "cassette_flip"
+	var/side1_icon = "cassette_flip"
+	var/side2_icon = "cassette_flip"
+	var/id = 1
+	var/list/song_names = list("side1" = list(),
+							   "side2" = list())
+
+	var/list/songs = list("side1" = list(),
+						  "side2" = list())
+
+/datum/cassette/cassette_tape/blank
+	name = "Blank Cassette"
+	desc = "A plastic cassette tape"
+	icon_state = "cassette_flip"
+	side1_icon = "cassette_flip"
+	id = 2
+	song_names = list("side1" = list("test",\
+									 "test2"),
+					  "side2" = list())
+
+	songs = list("side1" = list("https://www.youtube.com/watch?v=uChyiLL621Y&list=RDuChyiLL621Y&start_radio=1",\
```

Comment:
hmm not sure how i feel about having a youtube link embedded into code... but im not sure what an alternative is. I suppose json but that feels pointless for just a single link

Reply:
this was a debug one oops not meant to be there